exec/run-npm/src/index.ts

```diff
@@ -14,7 +14,9 @@ export function runNpm (npmPath: string | undefined, args: string[], options?: R
     cwd: options?.cwd ?? process.cwd(),
     stdio: 'inherit',
     userAgent: undefined,
-    env: options?.env ?? {},
+    env: process.env.COREPACK_ROOT ?
+      { ...options?.env, COREPACK_ENABLE_STRICT: '0' } :
+      options?.env ?? {},
```

Comment:
Thinking about it more, it's technically possible that `pnpm` would not be provided by Corepack, but `npm` would be.
```suggestion
    env: { ...options?.env, COREPACK_ENABLE_STRICT: '0' },
```